js/gdax.js

```diff
@@ -269,20 +276,39 @@ module.exports = class gdax extends Exchange {
                 'currency': market['quote'],
             };
         }
+        let type = undefined;
+        if ('liquidity' in trade) {
+            type = (trade['liquidity'] == 'T') ? 'Taker' : 'Maker';
+        }
         return {
-            'id': trade['trade_id'].toString (),
+            'id': trade['trade_id'].toString(),
+            'order': trade['order_id'].toString(),
             'info': trade,
             'timestamp': timestamp,
             'datetime': this.iso8601 (timestamp),
             'symbol': symbol,
-            'type': undefined,
+            'type': type,
             'side': side,
-            'price': parseFloat (trade['price']),
-            'amount': parseFloat (trade['size']),
+            'price': parseFloat(trade['price']),
```

Comment:
I don't like this style either, but [per CONTRIBUTING](https://github.com/ccxt/ccxt/blob/master/CONTRIBUTING.md#derived-exchange-classes), 

> every opening bracket like `(` or `{` should have a space before it!